self is not defined on pdf.worker

Link to PDF file (or attach file here):

Configuration:
- Web browser and its version: It's running on server-side with NodeJS (I tested v7.4 and LTS)
- Operating system and its version: macOS Sierra
- PDF.js version: 0.8.0
- Is an extension: Not. I'm only using the library. 

Steps to reproduce the problem:
1. Run pdfjs from server-side with NodeJS

What is the expected behavior? (add screenshot)
Compile normally. 
What went wrong? (add screenshot)
Don't compile. 

<pre><code>
/node_modules/pdfjs-dist/build/pdf.worker.js:19426
 var handler = new MessageHandler('worker', 'main', self);
                                                    ^

ReferenceError: self is not defined
    at initializeWorker (/Users/AKH000/Desktop/Projects/scripts/pdfs/node_modules/pdfjs-dist/build/pdf.worker.js:19426:53)
    at Object.<anonymous> (/Users/AKH000/Desktop/Projects/scripts/pdfs/node_modules/pdfjs-dist/build/pdf.worker.js:19431:2)
    at Object.QeTable.qe (/Users/AKH000/Desktop/Projects/scripts/pdfs/node_modules/pdfjs-dist/build/pdf.worker.js:19436:30)
    at __w_pdfjs_require__ (/Users/AKH000/Desktop/Projects/scripts/pdfs/node_modules/pdfjs-dist/build/pdf.worker.js:45:30)
    at Object.<anonymous> (/Users/AKH000/Desktop/Projects/scripts/pdfs/node_modules/pdfjs-dist/build/pdf.worker.js:50546:23)
    at __w_pdfjs_require__ (/Users/AKH000/Desktop/Projects/scripts/pdfs/node_modules/pdfjs-dist/build/pdf.worker.js:45:30)
    at globalScope (/Users/AKH000/Desktop/Projects/scripts/pdfs/node_modules/pdfjs-dist/build/pdf.worker.js:91:18)
    at /Users/AKH000/Desktop/Projects/scripts/pdfs/node_modules/pdfjs-dist/build/pdf.worker.js:94:10
    at webpackUniversalModuleDefinition (/Users/AKH000/Desktop/Projects/scripts/pdfs/node_modules/pdfjs-dist/build/pdf.worker.js:18:20)
    at Object.<anonymous> (/Users/AKH000/Desktop/Projects/scripts/pdfs/node_modules/pdfjs-dist/build/pdf.worker.js:25:3)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/AKH000/Desktop/Projects/scripts/kindleQuotes/helpers/script.js:5:17)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)

npm ERR! Darwin 16.1.0
npm ERR! argv "/Users/AKH000/.nvm/versions/node/v7.4.0/bin/node" "/Users/AKH000/.nvm/versions/node/v7.4.0/bin/npm" "start"
npm ERR! node v7.4.0
npm ERR! npm  v4.0.5
npm ERR! code ELIFECYCLE
npm ERR! Quotes@0.0.1 start: `node ./bin/www`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the Quotes@0.0.1 start script 'node ./bin/www'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the Quotes package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node ./bin/www
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs Quotes
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls Quotes
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/AKH000/Desktop/Projects/scripts/pdfs/npm-debug.log
</pre></code>

The file that use pdfjs contains this: 
<pre><code>var pdfjs = require("pdfjs-dist/build/pdf");
var workerSrc = require('pdfjs-dist/build/pdf.worker');
var cMapUrl = 'pdfjs-dist/cmaps';
function getTextContent(data){
  pdfjs.workerSrc = workerSrc;
  pdfjs.cMapUrl = cMapUrl;
  pdfjs.cMapPacked = true;
  var text = [];
  return pdfjs.getDocument(data).then(function(pdf) {
    var pages = [];
    for (var i = 0; i < pdf.numPages; i++) {
        pages.push(i);
    }
    return Promise.all(pages.map(function(pageNumber) {
      return pdf.getPage(pageNumber + 1).then(function(page) {
        return page.getTextContent().then(function(textContent) {
          return textContent.items.map(function(item) {
              return item;
          })
        });
      });
    })).then(function(pages){
      return pages;
    });
  }); //getDocument
}</pre></code>

I tried installing pdfjs-dist and it still not working, I tried compile by my self the library but it doesn't work. This actually it was working, but I don't know if the dependencie was updated or I'm forgeting something.

--- a/src/core/worker.js
+++ b/src/core/worker.js
@@ -12,7 +12,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-/* globals module */
+/* globals process, __pdfjsdev_webpack__ */
 
 'use strict';
 
@@ -1012,9 +1012,16 @@ function initializeWorker() {
   handler.send('ready', null);
 }
 
+function isNodeJS() {
+  // The if below protected by __pdfjsdev_webpack__ check from webpack parsing.
+  if (typeof __pdfjsdev_webpack__ === 'undefined') {
+    return typeof process === 'object' && process + '' === '[object process]';
+  }
+  return false;
+}
+
 // Worker thread (and not node.js)?
-if (typeof window === 'undefined' &&
-    !(typeof module !== 'undefined' && module.require)) {
+if (typeof window === 'undefined' && !isNodeJS()) {
   initializeWorker();
 }
 

